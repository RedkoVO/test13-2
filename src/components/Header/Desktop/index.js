import React from 'react'
import PropTypes from 'prop-types'

import HeaderMain from '../components/HeaderMain/Desktop'
import HeaderCatalog from '../components/HeaderCatalog/Desktop'
import HeaderGame from '../components/HeaderGame/Desktop'
import HeaderGames from '../components/HeaderGames/Desktop'

const Header = ({ type, children, handleLogout }) => (
  <React.Fragment>
    {type === 'catalog' && <HeaderCatalog />}
    {type === 'game' && <HeaderGame />}
    {type === 'games' && <HeaderGames children={children} />}
    {type === 'main' && <HeaderMain handleLogout={handleLogout} />}
  </React.Fragment>
)

Header.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node,
  handleLogout: PropTypes.func
}

export default Header
