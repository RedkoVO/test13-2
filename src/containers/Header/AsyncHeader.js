import Async from '../../components/App/components/Async'

export const AsyncHeader = Async({
  loader: () => import(/* webpackChunkName: "header_block" */ './index')
})

export default AsyncHeader
