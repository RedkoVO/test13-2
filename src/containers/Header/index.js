import compose from 'recompose/compose'
import { connect } from 'react-redux'
import { withHandlers, pure } from 'recompose'

import { resetApp } from '../../redux/actions/appReset'
import { getAllBalanceInfoSelector } from '../../selectors/balance'

import { clearStorageForLogout } from '../../utils/localStorageUtils'

import AsyncHeaderDesktop from '../../components/Header/Desktop'

const mapStateToProps = state => ({
  balance: getAllBalanceInfoSelector(state)
})

export default compose(
  connect(mapStateToProps),
  withHandlers({
    handleLogout: ({ history, dispatch }) => () => {
      dispatch(resetApp())
      clearStorageForLogout()
      history.push('/')
    }
  }),
  pure
)(AsyncHeaderDesktop)
