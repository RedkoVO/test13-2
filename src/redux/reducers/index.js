import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
