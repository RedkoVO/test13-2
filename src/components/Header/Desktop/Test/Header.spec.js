import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'

import Header from '../'

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders main', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Header type={'main'} handleLogout={() => {}} />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders game', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Header
              type={'game'}
              children={<div>Test</div>}
              handleLogout={() => {}}
            />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders games', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Header
              type={'games'}
              children={<div>Test</div>}
              handleLogout={() => {}}
            />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders catalog', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Header type={'catalog'} handleLogout={() => {}} />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
