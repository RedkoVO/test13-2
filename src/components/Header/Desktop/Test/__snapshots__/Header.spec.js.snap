// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders catalog 1`] = `
<div
  className="HeaderCatalog-root-30"
>
  <div
    className="HeaderCatalog-streamUserBlock-31"
  >
    <p
      className="HeaderCatalog-streamUserHello-32"
    >
      Hello, Ben A 
      <br />
      <a
        className="HeaderCatalog-streamUserVerification-33"
        href="/"
        onClick={[Function]}
      >
        KYC VERIFICATION
      </a>
    </p>
    <div
      className="HeaderCatalog-streamUserPhoto-34"
    >
      <img
        alt=""
        src="user.jpg"
      />
    </div>
  </div>
</div>
`;

exports[`renders game 1`] = `
<div
  className="HeaderGame-root-20"
>
  <div
    className="HeaderGame-gameUserBlock-21"
  >
    <p
      className="HeaderGame-gameUserHello-22"
    >
      Hello, Ben A 
      <br />
      <a
        className="HeaderGame-gameUserVerification-23"
        href="/"
        onClick={[Function]}
      >
        KYC VERIFICATION
      </a>
    </p>
    <div
      className="HeaderGame-gameUserPhoto-24"
    >
      <img
        alt=""
        src="user.jpg"
      />
    </div>
  </div>
</div>
`;

exports[`renders games 1`] = `
<div
  className="HeaderGames-root-25"
>
  <div
    className="HeaderGames-gamesUserBlock-26"
  >
    <p
      className="HeaderGames-gamesUserHello-27"
    >
       Hello, Ben A 
      <br />
      <a
        className="HeaderGames-gamesUserVerification-28"
        href="/"
        onClick={[Function]}
      >
        KYC VERIFICATION
      </a>
    </p>
    <div
      className="HeaderGames-gamesUserPhoto-29"
    >
      <img
        alt=""
        src="user.jpg"
      />
    </div>
  </div>
  <div>
    Test
  </div>
</div>
`;

exports[`renders main 1`] = `
<header
  className="HeaderMain-root-1"
>
  <div
    className="UserBlock-root-2"
  >
    <div
      className="UserBlock-user-3"
    >
      <div
        className="UserBlock-userBlock-4"
      >
        <p
          className="UserBlock-userHello-5"
        >
          Hello, anonymous 
          <br />
          <a
            className="UserBlock-userVerification-6"
            href="/"
            onClick={[Function]}
          >
            KYC VERIFICATION
          </a>
        </p>
        <div
          className="UserBlock-userPhoto-7"
        >
          <img
            alt=""
            src="user.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</header>
`;
